### DISCUSSION TITLE
Coral.AI Synology

### DISCUSSION THREAD

**Discussion by Davidvm86** [blakeblackshear/frigate#1685] [Ask A Question] [answered]
Hi,

I'm Running home assistant on a DS1621+ in the vmm of Synology DSM 6.2.
No I bought the Coral USB to use with frigate, but can't get it mounted.

If I mount the usb in the vmm and apply after opening the edit again screen it's unmounted automatically again.

Does anybody has this working on Synology?

**Comment by aptonline** [comment]
I have it working on a Synology DS918+ via docker and frigate, I know that beaderedtinker has recently purchased a Coral stick and has it working in a VM.  He’s likely to do a YouTube video on it shortly. https://twitter.com/beardedtinker/status/1433190082517913605?s=21

**Reply by Davidvm86** [reply]
Let try to update to DSM 7.0 hope that ik can get it to work then, in the vm.

**Reply by Floriszz** [reply]
@aptonline , can you share which image you took for DS918+ (=Intel Celeron J3455 based) from the list below?(ignoring the build version numbers of course)
![image](https://user-images.githubusercontent.com/19168937/143855833-538e3785-20a8-45c7-9253-d7e081a21e8c.png)

**Comment by Davidvm86** [comment, marked as answer]
Ok I Can confirm DSM 7.0 works with the Coral USB stick using the VMM for home assistant

**Reply by barosan909** [reply]
Hi David,
which usb ID it identifies with?  for me it is always the one below and due to that the frigate is not identifying it properly.
Bus 001 Device 004: ID 1a6e:089a Global Unichip Corp.

**Reply by NickM-27** [reply]
The USB changes IDs once it is initialized so you'll have to account for both

**Reply by barosan909** [reply]
In my case it stays 1a6e:089a, I have no clue how to initialize on DSM 7.

**Reply by NickM-27** [reply]
> In my case it stays 1a6e:089a, I have no clue how to initialize on DSM 7.

Assuming that is passed through, when you try to run it with frigate it will be initialized. Then the ID will change and frigate will crash, you must have it setup so both IDs (`Global Unicorp Chip` and `Google Inc.`) are passed through. 

Or the recommended way is to just have a PCIe USB card and pass that entire USB bus in

**Reply by barosan909** [reply]
"you must have it setup so both IDs (Global Unicorp Chip and Google Inc.) are passed through." How do I do that in VMM? Also when you say "when you try to run it with frigate it will be initialized" it does not do anything for me. It simply fails to recognize it, saying that no coral is detected(of course in the vm the 1a6e:089a id is visible when doing the lsusb. Btw i use the home assistant ova vm

**Reply by NickM-27** [reply]
When the USB has that ID and name `Global Unicorp Chip` it will not be recognized. It must have the other name / ID.

The general flow is:
1. USB has `ID 1a6e:089a Global Unichip Corp.`
2. Frigate starts and the USB is initialized, ID changes & name changes to `Google Inc.`
3. Frigate finds the Google Inc. device and runs. 

I don't know how to do it in VMM, just explaining what the issue is. There's lots of other discussions here with users showing how they solved this in Synology.

**Reply by barosan909** [reply]
This transition is noticed indeed when running it on RP. However with VMM i cannot have it as such.

**Reply by imne** [reply]
does it work on DSM 7.1.X also? considering to get a coral USB for my DS920+ ?

**Reply by barosan909** [reply]
I am running the latest version I have no luck to make it detectable by frigate. Also I notice, as some pointed out, that even at the DSM level it disapears fromt he lsusb completely after a while.... considering to make it work on my RP, there I noticed is still detected after a week.

**Reply by marcstyles** [reply]
In the same boat as you @barosan909 . I've set a passthrough `  devices:
          - /dev/bus/usb:/dev/bus/usb` in my docker compose but that doesn't seem to work. The NAS recognises the usb as _1a6e:089a Global Unichip Corp._ 
 
Not sure how @Davidvm86 got it to work but we could sure use his input 🙂

**Comment by weltenwort** [comment]
For anyone looking to get it working in docker on DSM7, I explained the solution that works for me in https://github.com/blakeblackshear/frigate/discussions/2426#discussioncomment-1901054.
